zip-reduce-shielded_tx_fee.rst

```diff
@@ -0,0 +1,135 @@
+in this document is in reference to Z to Z transaction fee paid
+to miners on the Zcash network for the work on including
+the shielded transaction in a block.
+
+
+Abstract
+========
+
+The goal of this ZIP is to gather wallet developers, miners & Zcash users
+for consensus on reducing the default transaction fees and
+to get the Zcash default transaction fee reduced from 10,000 zats
+to 1000 zats.
+
+With reduced fees, it will be cheaper to transact on the Zcash network,
+while also inviting novel use cases for privacy
+preserving applications that would benefit from the Privacy,
+Security and Programmable Money aspects of the Zcash chain.
+
+
+Out of Scope for this Proposal
+============
+
+Discussion around transaction fees for T to T, T to Z or Z to T.
+
+
+Motivation
+============
+
+The default transaction fee presently is 0.0001 ZEC or 10,000 zats.
+With ZEC market price of $100, a user can send 10,000 transactions
+for 1 ZEC, that turns out to 1 cent per transaction and it rises
+with the increase in the price of ZEC.
+
+With increase in light wallet adoptions on mobile clients, many users
+will be new to the Zcash eco-system. And the fact that the
+transaction fees are paid by the sender (not the receiver) is
+new information to users who might use Zcash for e-commerce
+and app interactions that might result in several transactions each day.
+
+Privacy must not cost a premium. The cost of 10 shielded transactions
+buys 1GB of mobile data `in India today <https://www.cable.co.uk/mobiles/worldwide-data-pricing/>`.
+
+Zcash users must be able to do more with their ZEC balance
+than worry about paying the premium for shielded transactions.
+
+
+Requirements for consensus
+-------------------------
+
+The change to the conventional transaction fees must be undertaken sooner
+as it gets difficult to gain consensus with the growth in the network
+of wallets, exchanges, miners and third parties involved.
+
+The following parties need to be part of the consensus:
+
+* A guarantee from mining groups is required to include the lowered default fee transactions in the next block.
+* Wallet developers need to provide commitment to update the software to use the new fee.
+* Zcash documentation and community outreach must be undertaken to make the change known.
+
+
+Security and privacy considerations
+-----------
+
+Unique transaction fees can cause linkability within transactions,
+hence this change must be accepted by majority, if not all popular
+shielded transaction software providers before announcing the change.
+
+Varying/unique fees are bad for privacy, for the short term before blocks get full,
+it’s fine for everyone to use a constant fee. [#nathan-1]_
+
+Long term, the issue of fees needs to be re-visited as the blocks start getting consistently full.
+And the possibility of a DoS scenario becomes critical in nature. New ZIP with flexible fees [#ian-1]_
+along with scaling solutions need to be evaluated and applied.
```

Comment:
```suggestion
New ZIPs with flexible fees, such as [#ian-1]_, along with scaling solutions need to be evaluated and applied.

Denial Of Service Vulnerability
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

A transaction-rate-based denial of service attack occurs when an attacker generates enough transactions over a window of time to prevent legitimate transactions from being mined, or to hinder syncing blocks for full nodes or miners.

There are two primary protections to this kind of attack in Zcash: the block size limit, and variable transaction fees. The block size limit ensures that full nodes and miners can sync blocks even if they are completely full. However it does not protect users sending legitimate transactions to have their transactions confirmed in a timely manner.

Variable fees can mitigate this kind of denial of service because in there are more transactions available than can fit into a single block, a miner is assumed to choose the transactions that pay the highest fees. If legitimate wallets increase their fees during this condition, the attacker must also increase the fees of their transactions. This imposes a growing and ongoing cost to the attacker which limits the time window they can continue the attack.

This proposal does not alter how fees are paid from transactions to miners. However, it does require wallets to use a fixed flat fee. Therefore during a transaction rate DoS attack, legitimate fees may not rise, so an attacker can extend an attacker for a longer window for the same cost.

This ZIP does not address this concern. A future ZIP should address this issue for shielded wallets.
```